### DISCUSSION TITLE
Had code signing issues following steps in Basic Setup docs

### DISCUSSION THREAD

**Discussion by skfoo** [sparkle-project/Sparkle#2539] [Ideas]
I just integrated Sparkle into an app following the documentation, but it wasn't code signed correctly and failed to launch with Library Validation enabled. I integrated using Swift Package Manager. To fix it, I copied the 'Code Signing Style' (automatic) and 'Development Team' from the target settings to the project-wide settings. There weren't any existing customized values for those.

It'd be helpful if that was in the setup guide.

Edit: Ugh, ok that didn't work, I still had Library Validation off when I tried that (－‸ლ) The actual fix is that I had to add @loader_path/../Frameworks to Runpath Search Paths; this is mentioned for the manual installation documentation but not for Swift Package Manager.

**Comment by zorgiepoo** [comment]
https://sparkle-project.org/documentation/

> If you enable Library Validation, which is part of the Hardened Runtime and required for notarization, you will also need to either sign your application with an Apple Development certificate for development (requires being in Apple’s developer program), or disable library validation for Debug configurations only. Otherwise, the system may not let your application load Sparkle if you attempt to sign to run locally via an ad-hoc signature. This is not an issue for distribution when you sign your application with a Developer ID certificate.

It is already there.

**Reply by skfoo** [reply]
That has nothing to do with what I wrote.

**Reply by zorgiepoo** [reply]
Yes it does, that is regarding signing identity used when library validation is enabled, or disabling library validation for debug otherwise.

New Xcode projects have `@executable_path/../Frameworks` for the runtime search paths as well by default which should be sufficient for regular apps. This is indeed noted in the manual step but it also means this typically does not require extra configuration. If you had needed this instruction to help you integrating a framework with SPM, this may mean you already started off on a wrong foot, or have a less than usual situation (maybe a plug-in?)

**Reply by zorgiepoo** [reply]
..also consider my situation where I get a bunch of questions from developers where many of them are not really even specific to Sparkle (like this one). It gets tedious at times.
